conda/switch_cuda_version.sh

```diff
@@ -17,11 +17,18 @@ mkdir -p /usr/local
 rm -fr /usr/local/cuda
 ln -s "$CUDA_DIR" /usr/local/cuda
 
+CUDA_VERSION_DOT=$(nvcc --version | sed -n 4p|cut -f5 -d" " | cut -f1 -d",")
+export CUDA_VERSION=${CUDA_VERSION_DOT/./}
 if [[ "$OSTYPE" == "msys" ]]; then
-    export CUDA_VERSION=`ls /usr/local/cuda/bin/cudart64*.dll | head -1 | tr '._' ' ' | cut -d ' ' -f2`
-    export CUDNN_VERSION=`ls /usr/local/cuda/bin/cudnn64*.dll | head -1 | tr '._' ' ' | cut -d ' ' -f2`
+    export CUDNN_VERSION=$(ls /usr/local/cuda/bin/cudnn64*.dll | head -1 | tr '._' ' ' | cut -d ' ' -f2)
+
+    # we want CUDNN_VERSION=8.1 for CUDA 11.2, not just 8
+    if [[ "$CUDA_VERSION" == '112' ]]; then
+        CUDNN_MAJOR=`grep 'define CUDNN_MAJOR' $(find /usr/local/cuda/ -name cudnn_version.h) | cut -d' ' -f3`
+        CUDNN_MINOR=`grep 'define CUDNN_MINOR' $(find /usr/local/cuda/ -name cudnn_version.h) | cut -d' ' -f3`
+        export CUDNN_VERSION=$CUDNN_MAJOR.$CUDNN_MINOR
```

Comment:
Also separate declare and assign here to avoid masking return values.
Is it better to check neither CUDNN_MAJOR nor CUDNN_MINOR is empty before the shell script proceeds?

Reply:
> Also separate declare and assign here to avoid masking return values.

Shellcheck doesn't warn about that here, my guess is because there's no way for the simple string concatenation to fail.